GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -198,28 +201,43 @@ private bool TryRunCleanUp(out string error)
         {
             error = null;
 
-            string errorMessage = null;
+            string errorMessage = string.Empty;
             if (!this.LaunchInsideSpinner(
                 () =>
                 {
-                    if (!this.TryReleaseUpgradeLock(out errorMessage))
+                    bool success = true;
+                    string unlockError;
+                    if (!this.TryReleaseUpgradeLock(out unlockError))
                     {
-                        return false;
+                        errorMessage += unlockError + Environment.NewLine;
+                        success = false;
                     }
 
-                    if (this.remount && !this.preRunChecker.TryMountAllGVFSRepos(out errorMessage))
+                    string remountError;
+                    if (this.remount && !this.preRunChecker.TryMountAllGVFSRepos(out remountError))
                     {
                         EventMetadata metadata = new EventMetadata();
                         metadata.Add("Upgrade Step", nameof(this.TryRunCleanUp));
-                        this.tracer.RelatedError(metadata, $"{nameof(this.preRunChecker.TryMountAllGVFSRepos)} failed. {errorMessage}");
-                        return false;
+                        this.tracer.RelatedError(metadata, $"{nameof(this.preRunChecker.TryMountAllGVFSRepos)} failed. {remountError}");

```

Comment:
I recommend adding `remountError` to `metadata` rather than including it in the error message, it will make the error easier to query in Kusto.